src/engine/QMLUrl.js

```diff
@@ -0,0 +1,28 @@
+function QMLUrl(val) {
```

Comment:
Not sure if this is the correct location for the new `url` type. Perhaps put it in `qtbase`?

/cc @akreuzkamp, @labsin — thoughts?